### PR TITLE
Implemented Button Loader

### LINKED ISSUE #23: Button Loader
if you hit submit and a API call is going to happen then button should have a loader and should be disable

### PR DESCRIPTION
**Pull request by rishabhrawat05**
---
name: Pull request
about: Maintain this format while making a PR
title: ""
labels: ""
assignees: ""
---

## Description

@Git21221 I have implemented the Button Loader Component in the project

Fixes #23 

## Type of Change

Please delete options that are not relevant:

- [ ] Bug fix (non-breaking change that fixes an issue)
- [Yes] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (changes to documentation only)
- [ ] Chore (cleanup or minor improvement that does not change functionality)

## Checklist

- [ Yes] My code follows the style guidelines of this project.
- [Yes ] I have performed a self-review of my code.
- [Yes ] I have commented my code, particularly in hard-to-understand areas.
- [No] I have made corresponding changes to the documentation.
- [ Yes] My changes generate no new warnings or errors.
- [Yes ] I have added tests that prove my fix is effective or that my feature works.
- [Yes ] New and existing unit tests pass locally with my changes.

## Screenshots (if applicable)

Screenshot (1)

## Additional Information

Please add following labels "hacktober-fest accepted, level label"

### CODE REVIEW

**Comment by Git21221** [comment]
@rishabhrawat05 change the branch from main to dev

File: src/components/Button/ButtonLoader.tsx
Comment (Git21221) [inline src/components/Button/ButtonLoader.tsx:1]: looks fine, can we modify existing Button component??

Comment (rishabhrawat05) [reply src/components/Button/ButtonLoader.tsx:1]: Can you explain more about it

**Comment by rishabhrawat05** [comment]
> @rishabhrawat05 change the branch from main to dev

@Git21221  updated

File: src/components/Button/ButtonLoader.tsx
Comment (Git21221) [reply src/components/Button/ButtonLoader.tsx:1]: there is a existing button component. If you check it you will understand  that you can merge your code with the Button component code.

Comment (Git21221) [reply src/components/Button/ButtonLoader.tsx:1]: @rishabhrawat05 let me know if you understand

Comment (rishabhrawat05) [reply src/components/Button/ButtonLoader.tsx:1]: So you are saying to remove button loader and add that functionality to button component itself

Comment (Git21221) [reply src/components/Button/ButtonLoader.tsx:1]: yes if it's possible, check once

Comment (rishabhrawat05) [reply src/components/Button/ButtonLoader.tsx:1]: Ok, then I will update the changes

Comment (Git21221) [reply src/components/Button/ButtonLoader.tsx:1]: yes, and make it optional field

Comment (rishabhrawat05) [reply src/components/Button/ButtonLoader.tsx:1]: Sure

**Comment by rishabhrawat05** [comment]
Hey! @Git21221 
Changes Implemented

File: src/components/Button/Button.tsx
Comment (Git21221) [inline src/components/Button/Button.tsx:1]: depending upon size of button, will the size of loader also change? I see it's hardcoded 20, is it for small or medium or large?

Comment (rishabhrawat05) [reply src/components/Button/Button.tsx:1]: I will change it, I forgot about resizing

Comment (rishabhrawat05) [reply src/components/Button/Button.tsx:1]: Hey! @Git21221 
I have checked the working of resizing of button with small, medium and large and loader is working fine with size 20

Comment (Git21221) [reply src/components/Button/Button.tsx:1]: @rishabhrawat05 ok fine

Comment (Git21221) [reply src/components/Button/Button.tsx:1]: @rishabhrawat05 is it possible if dev wants then put their custom animation logo?

Comment (rishabhrawat05) [reply src/components/Button/Button.tsx:1]: On button??

**Comment by rishabhrawat05** [comment]
@Git21221 
Please add labels hacktoberfest-accepted and a level label as this contribution will not be counted otherwise

File: src/components/Button/Button.tsx
Comment (Git21221) [reply src/components/Button/Button.tsx:1]: yes, if dev want they can replace the loading animation with custom animation component.

Comment (Git21221) [reply src/components/Button/Button.tsx:1]: take a parameter named "customLoadingComponent" and if the component presents then just replace with default one

Comment (rishabhrawat05) [reply src/components/Button/Button.tsx:1]: Like loadingAnimation="circular" like this and we can change it with any more alternative animation

**Pull request merged**